src/api/products/use-search-product.ts

```diff
@@ -0,0 +1,19 @@
+import type { AxiosError } from 'axios';
+import { createQuery } from 'react-query-kit';
+
+import { client } from '../common';
+import type { FetchProductsResponse } from './types';
+
+type Response = FetchProductsResponse;
+
+export const useSearchProduct = createQuery<
+  Response,
+  { product: string },
+  AxiosError
+>({
+  queryKey: ['searchProduct'],
+  fetcher: async ({ product }) => {
+    const { data } = await client.get(`products?text=${product}`);
+    return data;
+  },
+});
```

Comment:
It's not necessary to create another query, you can use the useProducts query, just pass the text parameter, that way you can also have search results paginated :)